lib/src/working_copy.rs

```diff
@@ -651,17 +651,34 @@ impl TreeState {
             git_ignore: base_ignores,
         }];
         trace_span!("traverse filesystem").in_scope(|| -> Result<(), SnapshotError> {
+            let (tree_builder_tx, tree_builder_rx) = channel();
```

Comment:
nit: `tree_builder_tx/rx` made me think that this channel was sending `TreeBuilder`s. How about `tree_entry_tx/rx`?

Reply:
Or `tree_entries_tx/rx` (plural) to match the other two